topoembedx/classes/hope.py

```diff
@@ -0,0 +1,174 @@
+        return_eigenval is True, eigenvalues are also returned alongside the positional encodings.
+        """
+        n = A.shape[0]
+        # Compute the degree matrix D^-0.5
+        D = sparse.diags(np.squeeze(np.asarray(np.power(np.sum(A, axis=1), -0.5))))
+
+        # Compute the Laplacian matrix L = I - D^-0.5 * A * D^-0.5
+        L = np.eye(A.shape[0]) - D @ A @ D
+
+        # Compute the eigenvectors of L
+        EigVal, EigVec = np.linalg.eig(L)
+
+        # Select the k smallest non-trivial eigenvectors
+        max_freqs = min(n - 1, k)
+        kpartition_indices = np.argpartition(EigVal, max_freqs)[: max_freqs + 1]
+        topk_eigvals = EigVal[kpartition_indices]
+        topk_indices = kpartition_indices[topk_eigvals.argsort()][1:]
+        topk_EigVec = EigVec[:, topk_indices]
+
+        # Randomly flip signs of the eigenvectors
+        rand_sign = 2 * (np.random.rand(max_freqs) > 0.5) - 1.0
+        PE = np.multiply(rand_sign, topk_EigVec.astype(np.float32))
+
+        if n <= k:
+            temp_EigVec = np.zeros((n, k - n + 1), dtype=np.float32)
+            PE = np.concatenate((PE, temp_EigVec), axis=1)
+            temp_EigVal = np.full(k - n + 1, np.nan, dtype=np.float32)
+            eigvals = np.concatenate((topk_eigvals, temp_EigVal), axis=0)
+        else:
+            eigvals = topk_eigvals
+
+        if return_eigenval:
+            return np.array(PE), eigvals
+
+        # Return the Laplacian positional encodings
+        return np.array(PE)
+
+    def fit(
+        self,
+        complex,
+        neighborhood_type="adj",
+        neighborhood_dim={"rank": 0, "to_rank": -1},
+    ):
+        """Fit a Higher Order Geometric Laplacian EigenMaps model.
+
+        Parameters
+        ----------
+        complex : TopoNetX object
+            A complex object. The complex object can be one of the following:
+            - CellComplex
+            - CombinatorialComplex
+            - ColoredHyperGraph
+            - SimplicialComplex
+            - PathComplex
+        neighborhood_type : str
+            The type of neighborhood to compute. "adj" for adjacency matrix, "coadj" for coadjacency matrix.
+        neighborhood_dim : dict
+            The dimensions of the neighborhood to use. If `neighborhood_type` is "adj", the dimension is
+            `neighborhood_dim['rank']`. If `neighborhood_type` is "coadj", the dimension is `neighborhood_dim['rank']`
+            and `neighborhood_dim['to_rank']` specifies the dimension of the ambient space.
+
+        Notes
+        -----
+        Here, neighborhood_dim={"rank": 1, "to_rank": -1} specifies the dimension for
+        which the cell embeddings are going to be computed.
+        rank=1 means that the embeddings will be computed for the first dimension.
+        The integer 'to_rank' is ignored and only considered
+        when the input complex is a combinatorial complex.
+
+        Examples
+        --------
+        import toponetx as tnx
```

Comment:
Missing >>>
Without it, it will not format on the doc website correctly.
See guidelines for writing docstrings